docs/docs/how-tos/encryption/custom_encryption.md

```diff
@@ -0,0 +1,451 @@
+            "my.customer.org/phone": "+1-555-0123",  # Encrypted
+        }
+    )
+    ```
+
+=== "Python RemoteGraph"
+
+    ```python
+    import base64
+    import json
+    from langgraph.pregel.remote import RemoteGraph
+
+    # Define encryption context
+    encryption_context = {
+        "tenant_id": "customer-123",
+        "key_id": "arn:aws:kms:us-east-1:123456789:key/abc-def-123"
+    }
+
+    # Encode as base64
+    encoded_context = base64.b64encode(
+        json.dumps(encryption_context).encode()
+    ).decode()
+
+    # Create remote graph with encryption context
+    remote_graph = RemoteGraph(
+        "agent",
+        url="http://localhost:2024",
+        headers={"X-Encryption-Context": encoded_context}
+    )
+
+    # The encryption context is automatically used for all operations
+    result = await remote_graph.ainvoke(
+        {"messages": [{"role": "user", "content": "Hello"}]},
+        config={"configurable": {"thread_id": "thread-1"}}
+    )
+    ```
+
+=== "CURL"
+
+    ```bash
+    # Encode encryption context
+    ENCRYPTION_CONTEXT=$(echo -n '{"tenant_id":"customer-123","key_id":"arn:aws:kms:us-east-1:123456789:key/abc-def-123"}' | base64)
+
+    # Create thread with encrypted metadata
+    curl -X POST http://localhost:2024/threads \
+      -H "Content-Type: application/json" \
+      -H "X-Encryption-Context: $ENCRYPTION_CONTEXT" \
+      -d '{
+        "metadata": {
+          "owner": "user-456",
+          "my.customer.org/email": "john@example.com",
+          "my.customer.org/phone": "+1-555-0123"
+        }
+      }'
+    ```
+
+## Important considerations
+
+### Searchability
+
+Fields with encrypted **values** cannot be reliably searched when using non-deterministic encryption (which most real-world encryption provides). In the example above:
+
+- ✅ **Can search**: `owner` field (unencrypted)
+- ❌ **Cannot search**: `my.customer.org/email` field (encrypted value)
+
+Design your metadata schema carefully:
+- Put searchable/filterable fields in unencrypted metadata
+- Put sensitive data that doesn't need to be searched in encrypted fields
+- Consider using prefixes to denote which fields should be encrypted
+
+### Decryption doesn't require re-passing context
+
+Once data is encrypted with a context, that context is stored alongside the encrypted data. When retrieving data, you **do not** need to pass the `X-Encryption-Context` header again - the stored context is automatically used for decryption.
+
+```python
+# Encrypt: pass context
+client = get_client(
+    url="http://localhost:2024",
+    headers={"X-Encryption-Context": encoded_context}
+)
+thread = await client.threads.create(metadata={"my.customer.org/secret": "value"})
+
+# Decrypt: no context needed
+client2 = get_client(url="http://localhost:2024")
+retrieved = await client2.threads.get(thread["thread_id"])
+# The secret field is automatically decrypted using the stored context
```

Comment:
This makes the encryption isolated from Auth and ownership (authN&authZ). Ideally, we would like to tie the encryption to the auth layers, so we can also trust that the client asking for decrypted data is also the owner of such data/resources